### PR TITLE
chore!: remove joda-time dependency

### LINKED ISSUE #409: Remove joda-time dependency
Can't you just use `java.time.LocalDateTime` instead? If you really need to use it, I suggest you mark the dependency as compile tome scope only and add some checks if the user has added joda-time in their classpath.

### PR DESCRIPTION
**Pull request by thinkingserious**
Removes joda.time dependency in favor of java.time

Fixes #409

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

File: src/test/java/com/twilio/http/RequestTest.java [hunk 1]
```diff
@@ -149,8 +151,8 @@ public void testAddQueryDateTimeRangeClosed() throws MalformedURLException {
     @Test
     public void testAddQueryDateTimeRangeClosedNotUTC() throws MalformedURLException {
         Request r = new Request(HttpMethod.GET, Domains.API.toString(), "/2010-04-01/foobar");
-        r.addQueryDateTimeRange("baz", Range.closed(new DateTime(2014, 1, 10, 14, 0, DateTimeZone.forID("America/Chicago")),
-            new DateTime(2014, 6, 1, 16, 0, DateTimeZone.forID("America/Chicago"))));
+        r.addQueryDateTimeRange("baz", Range.closed(ZonedDateTime.of(2014, 1, 10, 14, 0, 0, 0, ZoneId.of("America/Chicago")),
```
Comment (thinkingserious) [inline src/test/java/com/twilio/http/RequestTest.java:154]: I'm having trouble getting this test to pass. It appears that the `ZoneId` is not having any effect. I've tried several zones and get the same result.

Comment (childish-sambino) [reply src/test/java/com/twilio/http/RequestTest.java:154]: Add this to _addQueryDateTimeRange_: `.withZoneSameInstant(ZoneOffset.UTC)`

**Comment by childish-sambino** [comment]
Duplicate of https://github.com/twilio/twilio-java/pull/503

**Review by childish-sambino — APPROVED** [review]

**Comment by sullis** [comment]
I am adding an ArchUnit test for JodaTime: https://github.com/twilio/twilio-java/pull/590

**Pull request merged**